internal/events/aggregator.go

```diff
@@ -553,3 +591,44 @@ func (ag *aggregator) resolveBlobs(ctx context.Context, data fftypes.DataArray)
 	return true, nil
 
 }
+
+func (ag *aggregator) rewindForBlobArrival(ctx context.Context, blobHash *fftypes.Bytes32) error {
+
+	batchIDs := make(map[fftypes.UUID]bool)
```

Comment:
Should this just be an array instead of a map?

Reply:
The map is used to de-dup, as multiple messages might be associated.